metals/src/main/scala/scala/meta/internal/builds/ScalaCliBuildTool.scala

```diff
@@ -4,13 +4,46 @@ import java.security.MessageDigest
 
 import scala.concurrent.Future
 
+import scala.meta.internal.bsp.BspConfigGenerationStatus._
 import scala.meta.internal.metals.BuildInfo
 import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.internal.metals.UserConfiguration
 import scala.meta.internal.metals.scalacli.ScalaCli
 import scala.meta.io.AbsolutePath
 
-class ScalaCliBuildTool(val workspaceVersion: Option[String])
-    extends BuildTool {
+class ScalaCliBuildTool(
+    val workspaceVersion: Option[String],
+    userConfig: () => UserConfiguration,
+) extends BuildTool
+    with BuildServerProvider {
+
+  lazy val runScalaCliCommand: Option[Seq[String]] =
+    ScalaCli.localScalaCli(userConfig())
+
+  def createBspConfigIfNone(
+      workspace: AbsolutePath,
+      systemProcess: List[String] => Future[BspConfigGenerationStatus],
+  ): Future[BspConfigGenerationStatus] = {
+    if (ScalaCliBuildTool.pathsToScalaCliBsp(workspace).exists(_.isFile))
+      Future.successful(Generated)
+    else generateBspConfig(workspace, systemProcess)
+  }
+
+  override def createBspFileArgs(workspace: AbsolutePath): List[String] =
+    runScalaCliCommand.getOrElse(Seq("scala-cli")).toList ++ List(
+      "setup-ide",
+      workspace.toString(),
+    )
+
+  override def workspaceSupportsBsp(workspace: AbsolutePath): Boolean = {
+    val supports = runScalaCliCommand.nonEmpty
+    if (!supports) {
```

Comment:
Is that correct that if no scala cli is found locally we will not set up the project for a user? Could we not use the JVM launcher in that case?

Reply:
If we have `project.scala` I would create the bsp config since we currently assume that only Scala CLI projects will have it.

>  It seemed to me like if people decide to use scala-cli they would have it but maybe it's an incorrect assumption

Potentially, people could want to check out their project without setting scala-cli up first and we should make sure it works.

Reply:
If we have this fallback I can delete installing `scala-cli` from CI.

Reply:
How does it interact with using scala-cli as fallback? We will not create the .bsp file in that case, right? Unless we have `project.scala`? I thought that this issue was more related tot he other PR

Reply:
Yes, it will just start `scala-cli` as fallback w/o creating `bsp` config, unless you have `project.scala`. It that case it will always create `.bsp.scala-cli`. Which other PR? Basically it should all work both w/ and w/o installed `scala-cli` it will just test a bit different parts of code.

Reply:
I meant this one https://github.com/scalameta/metals/pull/5221 <- I totally forgot this was already merged.